plugins/eni/engine/engine.go

```diff
@@ -304,9 +317,28 @@ func (engine *engine) DoesMACAddressMapToIPV6Address(macAddress string, ipv6Addr
 
 func (engine *engine) doesMACAddressMapToIPAddress(macAddress string, addressToFind string, metatdataPathSuffix string) (bool, error) {
 	// TODO Use fmt.Sprintf and wrap that in a method
-	addressesResponse, err := engine.metadata.GetMetadata(metadataNetworkInterfacesPath + macAddress + metatdataPathSuffix)
+	var addressesResponse string
+	var err error
+
+	attempts := 1
+	for {
```

Comment:
You're just checking for `err != nil` outside this loop, so why not just make this a for loop with the count in the loop instead of the body?

```go
for attempts := 0; attempts < engine.metadataMaxRetryCount; attempts++
```

Reply:
The reason for doing it this way is that I don't want an additional sleep on the last attempt if there's an error.  `for attempts := 0; attempts < engine.metadataMaxRetryCount; attempts++` doesn't give me that. Open to suggestions here.

Reply:
Got it, that makes sense.